---
title: "GWAS variants"
output: html_notebook
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(rentrez)
library(GEOmetadb)
library(RSQLite)
library(magrittr)
library(dplyr)
library(tidyr)
devtools::load_all('../')
```

```{r}
quilted.arrhythmia = data.table::fread('../data/CVDs/arrhythmia.tsv', sep='\t')
quilted.cm = data.table::fread('../data/CVDs/cardiomyopathy.tsv', sep='\t')
```

```{r}
testVar = 'rs74315379' # multiple conflicting clinical significance
variations = quilted.cm$Variation %>% unique()
quilted.cm%>%
    `[`(.$Variation == testVar)
```

```{r}

quilted.cm[quilted.cm$Variation == 'rs1064792925',]
gwas = data.table::fread('../data/gwas_catalog_v1.0.2-associations_e93_r2018-08-28.tsv', sep='\t')
gwas$`DISEASE/TRAIT` %>% unique() %>% length()
```

```{r}
gwas.mi = gwas[grep('[mM]yocardial infarction', gwas$`DISEASE/TRAIT`),] %>%
    `$<-`('DATE', as.Date(.$DATE))
ggplot(gwas.mi) +
    geom_point(aes(x=DATE,y=PVALUE_MLOG, color=SNPS)) +
    theme(legend.position = 'none')
```

## Dilated myocardiopathy

### GWAS

```{r}
gwas.dcm = gwas[grep('[dD]ilated [cC]ardiomyopathy', gwas$`DISEASE/TRAIT`),] %>%
    `$<-`('DATE', as.Date(.$DATE))
gwas.dcm$`REPORTED GENE(S)`  %>% sapply(strsplit, split = ',') %>% do.call(c, .) %>% trimws() %>% unique()
```

### Ensembl

```{r}
dcm.dominant = data.table::fread('../data/dilated-cardiomyopathy-dominant-loci-ensembl.csv', sep=',')
dcm.recessive = data.table::fread('../data/dilated-cardiomyopathy-recessive-loci-ensembl.csv', sep=',')
dcm.loci = rbind(dcm.dominant, dcm.recessive) %>%
    `$<-`('inheritance', c(rep('dominant', nrow(dcm.dominant)), rep('recessive', nrow(dcm.recessive))))
    
```

The number of genes associated with
* _Dilated cardiomyopathy, dominant_ is `r dcm.dominant[['Reported gene(s)']] %>% trimws() %>% sapply(strsplit, split = '\r') %>% do.call(c,.) %>% as.character() %>% unique() %>% length()`
* _Dilated cardiomyopathy, recessive is `r dcm.recessive[['Reported gene(s)']] %>% trimws() %>% sapply(strsplit, split = '\r') %>% do.call(c,.) %>% as.character() %>% unique() %>% length()`
* _Dilated cardiomyopathy (both) is `r dcm.loci[['Reported gene(s)']] %>% trimws() %>% sapply(strsplit, split = '\r') %>% do.call(c,.) %>% as.character() %>% unique() %>% length()`

### disgenet

```{r}
dgn.mappings = data.table::fread('../data/disgenet/disease_mappings.tsv', sep='\t')
standardNames = data.table::fread('../data/conditionList.tsv', sep='\t') %>%
    merge(dgn.mappings[,c('diseaseId', 'code', 'name')], by.x = 'HPOTermId', by.y= 'code', all.x = TRUE, all.y = FALSE, sort = FALSE)
```

All the associations with CVDs found in Disgenet

```{r}
dgn.cvdIds = standardNames %>%
    `$`('diseaseId') %>%
    unique() %>% subset(., !is.na(.)) %>%
    data.frame('diseaseId' = .) %>%
    merge(dgn.mappings[,c('diseaseId', 'name')], all.x = TRUE, all.y = FALSE, sort=FALSE) %>% unique()
dgn.CVDs = data.table::fread('../data/disgenet/all_gene_disease_associations.tsv') %>%
    subset(diseaseId %in% dgn.cvdIds$diseaseId)
```

#### Gene association similarity among the diseases

```{r}
m = matrix(0, nrow=nrow(dgn.cvdIds),  ncol=nrow(dgn.cvdIds)) %>%
    set_rownames(dgn.cvdIds$name) %>%
    set_colnames(dgn.cvdIds$name)
for (i in 1:nrow(m)) {
    for (j in i:ncol(m)) {
        associations_i = subset(dgn.CVDs, diseaseId %in% dgn.cvdIds[i,'diseaseId']) %>% `$`('geneId') %>% unique()
        if (i == j) m[i,j] = length(associations_i)
        else {
            associations_j = subset(dgn.CVDs, diseaseId %in% dgn.cvdIds[j, 'diseaseId']) %>% `$`('geneId') %>% unique()
            m[i,j] = m[j,i] = intersect(associations_i, associations_j) %>% length()
        }
    }
}

cl = hclust(as.dist(-m))
cl$order
p = plot.heatmap(m[cl$order, cl$order]) +
    theme(axis.text.y = element_text(size = 18),
          axis.text.x = element_text(size = 16),
          legend.title = element_text(size = 18),
          legend.text = element_text(size = 18)) +
    guides(fill = guide_legend('Number of genes')) +
    scale_fill_gradient(low='lightblue', high='darkblue')
p
ggsave(filename = '../figures/disgenet-intersection.png', device = 'png', width = 18, height = 18)
```
