---
title: "Gene - Disease associations"
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(rentrez)
library(GEOmetadb)
library(RSQLite)
library(magrittr)
library(dplyr)
library(tidyr)
devtools::load_all('../')
```

## Exploratory analysis of GWAS

### Data files and structures

The GWAS-provided table of `studies` contains the association of a given phenotype in a given study (represented by Pubmed ID). Thus a single pubmed id (a publication) may  have multiple rows in this table, if it reports associations with multiple phenotypes.


```{r}
gwas = data.table::fread('../data/gwas/gwas_catalog_v1.0.2-associations_e93_r2018-08-28.tsv', sep='\t')
gwas.studies = data.table::fread('../data/gwas/gwas_catalog_v1.0.2-studies_r2018-09-15.tsv', sep='\t')
test.pmid = gwas.studies %>% count(PUBMEDID) %>% filter(n > 1) %>% `[[`(1,'PUBMEDID')
```

For example the study `r test.pmid` has `r nrow(filter(gwas.studies, PUBMEDID == test.pmid))`, each has a different `MAPPED_TRAIT`,

```{r}
filter(gwas.studies, PUBMEDID == test.pmid)
```

Each of such unique pair (PMID - MAPPED_TRAIT) is given a study accession uniquely identify the record within GWAS database.

```{r}
nrow(gwas.studies) == gwas.studies$`STUDY ACCESSION` %>% unique() %>% length()
```

The `associations` table report one _association_ per row. There are three types of associations.
1. The single SNP - trait association
2. The multiple-SNP haplotype - trait association, in which SNPs are separated by `;`
3. The SNP-SNP interaction, in which the pairs are described as `snp1 x snp2`

Some examples of multiple-SNP haplotypes

```{r}
gwas[grepl(pattern = ";", gwas$SNPS), c('SNPS', 'MAPPED_TRAIT', 'PVALUE_MLOG')]
```
Some examples of SNP interaction studies

```{r}
gwas[grepl(pattern = " x ", gwas$SNPS), c('SNPS', 'MAPPED_TRAIT', 'PVALUE_MLOG') ]
```

There are some SNPs that are not given a name with the pattern `rsXXXXX`, for example

```{r}
gwas[!grepl(pattern = 'rs\\d+', gwas$SNPS, perl=TRUE), 'SNPS']
```

### SNPs reported both as single SNP and in the haplotypes

There are SNPs associated with traits both as an individual variant and in linkage with other variants. The fact that a single SNP is associated with a trait does not exclude it from the linkage with other SNPs, but it is simply the result of single-SNP focused analyses.

```{r}
# start = proc.time()
type1 = gwas[(!grepl(pattern = ';', gwas$SNPS) & !grepl(pattern = ' x ', gwas$SNPS)),]$SNPS %>% unique()
type2 = gwas[grepl(pattern = ';', gwas$SNPS),]$SNPS %>% unique()
type3 = gwas[grepl(pattern = ' x ', gwas$SNPS),]$SNPS %>% unique()
singleInMulti = sapply(type1, FUN = function(y) {return(grep(pattern = paste0('\\b', y, '\\b'), x = type2, value = TRUE, perl = TRUE))})
# singleInInter = sapply(type1, FUN = function(y) {return(grep(pattern = paste0('\\b', y, '\\b'), x = type3, value = TRUE, perl = TRUE))})
# dt = proc.time() - start
# print(dt)
cnt.singleInMulti = sapply(singleInMulti, FUN = length)
tmp<-singleInMulti[which(cnt.singleInMulti > 1)] %>%
    do.call(c,.) %>%
    c(names(singleInMulti[which(cnt.singleInMulti > 1)])) %>%
    as.character()
dplyr::filter(gwas, SNPS %in% tmp)[c('SNPS', 'DISEASE/TRAIT')]
```

### Contexts of the variants

```{r}
split_records <- function(x, cols, split = ';') {
    lapply(1:nrow(x), FUN = function(i) {
        lapply(cols, function(j) {
            return(strsplit(x[[j]], split = split) %>% unlist() %>% trimws())
        }) %>%
            do.call(cbind, .) %>%
            return()
    }) %>%
        do.call(rbind, .) %>%
        data.frame() %>%
        set_names(cols) %>%
        dplyr::distinct() %>%
        return()
}

type1 = gwas[(!grepl(pattern = ';', gwas$SNPS) & !grepl(pattern = ' x ', gwas$SNPS)),c('SNPS', 'CONTEXT')]
type2 = gwas[grepl(pattern = ';', gwas$SNPS),c('SNPS', 'CONTEXT')]
type3 = gwas[grepl(pattern = ' x ', gwas$SNPS),c('SNPS', 'CONTEXT')]

variant.contexts = split_records(type2, cols = c('SNPS', 'CONTEXT'), split = ';') %>%
    rbind(split_records(type3, cols = c('SNPS', 'CONTEXT'), split = ' x ')) %>%
    rbind(type1) %>%
    dplyr::distinct()

# sanity check -- one variant, multiple contexts
dplyr::count(variant.contexts, SNPS)

dplyr::count(variant.contexts, CONTEXT) %>%
    ggplot(mapping =  aes(x = CONTEXT, y = n)) +
    geom_bar(stat='identity') + coord_flip()

count(gwas, CONTEXT)
gwas[gwas$CONTEXT == "3_prime_UTR_variant; intron_variant; intergenic_variant",]
setdiff(gwas.studies$PUBMEDID %>% unique(),gwas$PUBMEDID %>% unique())
gwas$`DISEASE/TRAIT` %>% unique() %>% length()
gwas$CONTEXT %>%  unique()  %>% length()
```

### Case study: `rs10454444`

```{r}
gwas[grepl(pattern = '\\brs10454444\\b', gwas$SNPS), c('SNPS', 'STRONGEST SNP-RISK ALLELE', 'RISK ALLELE FREQUENCY', 'PUBMEDID', 'STUDY', 'DISEASE/TRAIT', 'INITIAL SAMPLE SIZE', 'REPLICATION SAMPLE SIZE', 'REPORTED GENE(S)', 'MAPPED_GENE', 'PVALUE_MLOG')]
```

```{r}
gwas[gwas$PUBMEDID == '25948407',c('SNPS', 'STRONGEST SNP-RISK ALLELE', 'RISK ALLELE FREQUENCY', 'DISEASE/TRAIT', 'INITIAL SAMPLE SIZE', 'REPLICATION SAMPLE SIZE', 'REPORTED GENE(S)', 'MAPPED_GENE', 'PVALUE_MLOG') ]
```



```{r}
gwas.mi = gwas[grep('[mM]yocardial infarction', gwas$`DISEASE/TRAIT`),] %>%
    `$<-`('DATE', as.Date(.$DATE)) %>%
    `$<-`('RISK_ALLELE_FREQUENCY', as.numeric(.$`RISK ALLELE FREQUENCY`))
ggplot(gwas.mi) +
    geom_point(mapping = aes_string(x='PVALUE_MLOG',y='RISK_ALLELE_FREQUENCY', color='DATE')) +
    theme(legend.position = 'none')
count(gwas.mi, CONTEXT)
```

```{r}
dgn.mappings = data.table::fread('../data/disgenet/disease_mappings.tsv', sep='\t')
standardNames = data.table::fread('../data/conditionList.tsv', sep='\t') %>%
    merge(dgn.mappings[,c('diseaseId', 'code', 'name')], by.x = 'HPOTermId', by.y= 'code', all.x = TRUE, all.y = FALSE, sort = FALSE)
```

## Dilated myocardiopathy

### GWAS

```{r}
gwas.dcm = gwas[grep('[dD]ilated [cC]ardiomyopathy', gwas$`DISEASE/TRAIT`),] %>%
    `$<-`('DATE', as.Date(.$DATE))
gwas.dcm$`REPORTED GENE(S)`  %>% sapply(strsplit, split = ',') %>% do.call(c, .) %>% trimws() %>% unique()
```

### Ensembl

```{r}
dcm.dominant = data.table::fread('../data/dilated-cardiomyopathy-dominant-loci-ensembl.csv', sep=',')
dcm.recessive = data.table::fread('../data/dilated-cardiomyopathy-recessive-loci-ensembl.csv', sep=',')
dcm.loci = rbind(dcm.dominant, dcm.recessive) %>%
    `$<-`('inheritance', c(rep('dominant', nrow(dcm.dominant)), rep('recessive', nrow(dcm.recessive))))
    
```

The number of genes associated with
* _Dilated cardiomyopathy, dominant_ is `r dcm.dominant[['Reported gene(s)']] %>% trimws() %>% sapply(strsplit, split = '\r') %>% do.call(c,.) %>% as.character() %>% unique() %>% length()`
* _Dilated cardiomyopathy, recessive is `r dcm.recessive[['Reported gene(s)']] %>% trimws() %>% sapply(strsplit, split = '\r') %>% do.call(c,.) %>% as.character() %>% unique() %>% length()`
* _Dilated cardiomyopathy (both) is `r dcm.loci[['Reported gene(s)']] %>% trimws() %>% sapply(strsplit, split = '\r') %>% do.call(c,.) %>% as.character() %>% unique() %>% length()`

## GWAS

### Mapping disease names

```{r}
gwas.queries <- data.table::fread('../data/conditionList.tsv', sep='\t')$query %>% unique() %>%
    sapply(FUN = function(y) {
        gwas[grepl(pattern = y, x = gwas$MAPPED_TRAIT, ignore.case = TRUE) | grepl(pattern = y, x = gwas$`DISEASE/TRAIT`, ignore.case = TRUE), 'MAPPED_TRAIT'] %>%
            return()
    })

dgn.mappings = data.table::fread('../data/disgenet/disease_mappings.tsv', sep='\t')
standardNames = data.table::fread('../data/conditionList.tsv', sep='\t') %>%
    merge(dgn.mappings[,c('diseaseId', 'code', 'name')], by.x = 'HPOTermId', by.y= 'code', all.x = TRUE, all.y = FALSE, sort = FALSE)
```

### All the associations with CVDs


### Gene association similarity among the diseases

```{r}
m = matrix(0, nrow=nrow(dgn.cvdIds),  ncol=nrow(dgn.cvdIds)) %>%
    set_rownames(dgn.cvdIds$name) %>%
    set_colnames(dgn.cvdIds$name)
for (i in 1:nrow(m)) {
    for (j in i:ncol(m)) {
        associations_i = subset(dgn.CVDs, diseaseId %in% dgn.cvdIds[i,'diseaseId']) %>% `$`('geneId') %>% unique()
        if (i == j) m[i,j] = length(associations_i)
        else {
            associations_j = subset(dgn.CVDs, diseaseId %in% dgn.cvdIds[j, 'diseaseId']) %>% `$`('geneId') %>% unique()
            m[i,j] = m[j,i] = intersect(associations_i, associations_j) %>% length()
        }
    }
}

cl = hclust(as.dist(-m))
cl$order
p = plot.heatmap(m[cl$order, cl$order]) +
    theme(axis.text.y = element_text(size = 18),
          axis.text.x = element_text(size = 16),
          legend.title = element_text(size = 18),
          legend.text = element_text(size = 18)) +
    guides(fill = guide_legend('Number of genes')) +
    scale_fill_gradient(low='lightblue', high='darkblue')
p
ggsave(filename = '../figures/disgenet-intersection.png', device = 'png', width = 18, height = 18)
```

## disgenet

### Mapping disease names to database unique index

```{r}
dgn.mappings = data.table::fread('../data/disgenet/disease_mappings.tsv', sep='\t')
standardNames = data.table::fread('../data/conditionList.tsv', sep='\t') %>%
    merge(dgn.mappings[,c('diseaseId', 'code', 'name')], by.x = 'HPOTermId', by.y= 'code', all.x = TRUE, all.y = FALSE, sort = FALSE)
```

### All the associations with CVDs found in Disgenet

```{r}
dgn.cvdIds = standardNames %>%
    `$`('diseaseId') %>%
    unique() %>% subset(., !is.na(.)) %>%
    data.frame('diseaseId' = .) %>%
    merge(dgn.mappings[,c('diseaseId', 'name')], all.x = TRUE, all.y = FALSE, sort=FALSE) %>% unique()
dgn.CVDs = data.table::fread('../data/disgenet/all_gene_disease_associations.tsv') %>%
    subset(diseaseId %in% dgn.cvdIds$diseaseId)
```

### Gene association similarity among the diseases

```{r}
m = matrix(0, nrow=nrow(dgn.cvdIds),  ncol=nrow(dgn.cvdIds)) %>%
    set_rownames(dgn.cvdIds$name) %>%
    set_colnames(dgn.cvdIds$name)
for (i in 1:nrow(m)) {
    for (j in i:ncol(m)) {
        associations_i = subset(dgn.CVDs, diseaseId %in% dgn.cvdIds[i,'diseaseId']) %>% `$`('geneId') %>% unique()
        if (i == j) m[i,j] = length(associations_i)
        else {
            associations_j = subset(dgn.CVDs, diseaseId %in% dgn.cvdIds[j, 'diseaseId']) %>% `$`('geneId') %>% unique()
            m[i,j] = m[j,i] = intersect(associations_i, associations_j) %>% length()
        }
    }
}

cl = hclust(as.dist(-m))
cl$order
p = plot.heatmap(m[cl$order, cl$order]) +
    theme(axis.text.y = element_text(size = 18),
          axis.text.x = element_text(size = 16),
          legend.title = element_text(size = 18),
          legend.text = element_text(size = 18)) +
    guides(fill = guide_legend('Number of genes')) +
    scale_fill_gradient(low='lightblue', high='darkblue')
p
ggsave(filename = '../figures/disgenet-intersection.png', device = 'png', width = 18, height = 18)
```
