---
title: "Gene - Disease associations"
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(ggplot2)
library(rentrez)
library(GEOmetadb)
library(RSQLite)
library(magrittr)
library(dplyr)
library(tidyr)
devtools::load_all('../')
VAR.CONSEQUENCES = data.table::fread('../data/ensembl-vep-consequences.tsv', sep='\t')
```

## Exploratory analysis of GWAS

### Data files and structures

The GWAS-provided table of `studies` contains the association of a given phenotype in a given study (represented by Pubmed ID). Thus a single pubmed id (a publication) may  have multiple rows in this table, if it reports associations with multiple phenotypes.


```{r}
gwas = data.table::fread('../data/gwas/gwas_catalog_v1.0.2-associations_e93_r2018-08-28.tsv', sep='\t')
gwas.studies = data.table::fread('../data/gwas/gwas_catalog_v1.0.2-studies_r2018-09-15.tsv', sep='\t')
test.pmid = gwas.studies %>% count(PUBMEDID) %>% filter(n > 1) %>% `[[`(1,'PUBMEDID')
```

For example the study `r test.pmid` has `r nrow(filter(gwas.studies, PUBMEDID == test.pmid))`, each has a different `MAPPED_TRAIT`,

```{r}
filter(gwas.studies, PUBMEDID == test.pmid)
```

Each of such unique pair (PMID - MAPPED_TRAIT) is given a study accession uniquely identify the record within GWAS database.

```{r}
nrow(gwas.studies) == gwas.studies$`STUDY ACCESSION` %>% unique() %>% length()
```

The `associations` table report one _association_ per row. There are three types of associations.
1. The single SNP - trait association
2. The multiple-SNP haplotype - trait association, in which SNPs are separated by `;`
3. The SNP-SNP interaction, in which the pairs are described as `snp1 x snp2`

```{r}
type1.ind = (!grepl(pattern = ';', gwas$SNPS) & !grepl(pattern = ' x ', gwas$SNPS))
type2.ind = grepl(pattern = ';', gwas$SNPS)
type3.ind = grepl(pattern = ' x ', gwas$SNPS)

is.type1 = function(X, test_col = 'SNPS') { return(!grepl(pattern = ';', X[[test_col]]) & !grepl(pattern = ' x ', X[[test_col]])) }
is.type2 = function(X, test_col = 'SNPS') { return(grepl(pattern = ';', X[[test_col]])) }
is.type3 = function(X, test_col = 'SNPS') { return(grepl(pattern = ' x ', X[[test_col]])) }
```

Some examples of multiple-SNP haplotypes

```{r}
gwas[grepl(pattern = ";", gwas$SNPS), c('SNPS', 'MAPPED_TRAIT', 'PVALUE_MLOG')]
```
Some examples of SNP interaction studies

```{r}
gwas[grepl(pattern = " x ", gwas$SNPS), c('SNPS', 'MAPPED_TRAIT', 'PVALUE_MLOG') ]
```

There are some SNPs that are not given a name with the pattern `rsXXXXX`, for example

```{r}
gwas[!grepl(pattern = 'rs\\d+', gwas$SNPS, perl=TRUE), 'SNPS']
```

### SNPs reported both as single SNP and in the haplotypes

There are SNPs associated with traits both as an individual variant and in linkage with other variants. The fact that a single SNP is associated with a trait does not exclude it from the linkage with other SNPs, but it is simply the result of single-SNP focused analyses.

```{r}
# start = proc.time()
type1.snp = gwas[type1.ind,]$SNPS %>% unique()
type2.snp = gwas[type2.ind,]$SNPS %>% unique()
type3.snp = gwas[type3.ind,]$SNPS %>% unique()
singleInMulti = sapply(type1.snp, FUN = function(y) {return(grep(pattern = paste0('\\b', y, '\\b'), x = type2.snp, value = TRUE, perl = TRUE))})
# singleInInter = sapply(type1, FUN = function(y) {return(grep(pattern = paste0('\\b', y, '\\b'), x = type3, value = TRUE, perl = TRUE))})
# dt = proc.time() - start
# print(dt)
cnt.singleInMulti = sapply(singleInMulti, FUN = length)
tmp<-singleInMulti[which(cnt.singleInMulti > 1)] %>%
    do.call(c,.) %>%
    c(names(singleInMulti[which(cnt.singleInMulti > 1)])) %>%
    as.character()
dplyr::filter(gwas, SNPS %in% tmp)[c('SNPS', 'DISEASE/TRAIT')]
```

### Contexts of the variants

```{r}
split_records <- function(x, cols, split = ';') {
    lapply(1:nrow(x), FUN = function(i) {
        lapply(cols, function(j) {
            return(strsplit(x[[j]], split = split) %>% unlist() %>% trimws())
        }) %>%
            do.call(cbind, .) %>%
            return()
    }) %>%
        do.call(rbind, .) %>%
        data.frame() %>%
        set_names(cols) %>%
        dplyr::distinct() %>%
        return()
}

context1 = gwas[is.type1(gwas, 'SNPS'),c('SNPS', 'CONTEXT')]
context2 = gwas[is.type2(gwas, 'SNPS'),c('SNPS', 'CONTEXT')]
context3 = gwas[is.type3(gwas, 'SNPS'),c('SNPS', 'CONTEXT')]

# variant.contexts = split_records(context2, cols = c('SNPS', 'CONTEXT'), split = ';') %>%
    # rbind(split_records(context3, cols = c('SNPS', 'CONTEXT'), split = ' x ')) %>%
    # rbind(context1) %>%
    # dplyr::distinct()

variant.contexts = context1 %>% dplyr::distinct()

# one variant, multiple contexts
# dplyr::count(variant.contexts, SNPS)

context.cnt = dplyr::count(variant.contexts, CONTEXT)
```




#### Distribution of variants in genome

```{r}
gwas.queries <- data.table::fread('../data/conditionList.tsv', sep='\t')$query %>% unique() %>%
    lapply(FUN = function(y) {
        gwas[grepl(pattern = y, x = gwas$MAPPED_TRAIT, ignore.case = TRUE) | grepl(pattern = y, x = gwas$`DISEASE/TRAIT`, ignore.case = TRUE),] %>%
            return()
    })
gwas.CVDs <- do.call(rbind, gwas.queries)
# is.type1(gwas.CVDs) %>% sum() +
#     is.type2(gwas.CVDs) %>% sum() +
#     is.type3(gwas.CVDs) %>% sum()

context1 = gwas.CVDs[is.type1(gwas.CVDs, 'SNPS'),c('SNPS', 'CONTEXT')]
context2 = gwas.CVDs[is.type2(gwas.CVDs, 'SNPS'),c('SNPS', 'CONTEXT')]
context3 = gwas.CVDs[is.type3(gwas.CVDs, 'SNPS'),c('SNPS', 'CONTEXT')]

# variant.contexts = 
#     split_records(context2, cols = c('SNPS', 'CONTEXT'), split = ';') %>%
#     rbind(split_records(context3, cols = c('SNPS', 'CONTEXT'), split = ' x ')) %>%
#     rbind(context1) %>%
#     dplyr::distinct()

variant.contexts = context1 %>% dplyr::distinct()

context.cnt.cvd = dplyr::count(variant.contexts, CONTEXT) 
context.cnt.combined = 
    context.cnt.cvd %>%
    dplyr::full_join(x = context.cnt, y = ., all = TRUE, by = 'CONTEXT', incomparables = 0) %>%
    `[<-`(is.na(.[[2]]), 2, 0) %>%
    `[<-`(is.na(.[[3]]), 3, 0) %>%
    set_names(c('SOTerm', 'All traits', 'CVD')) %>%
    dplyr::left_join(VAR.CONSEQUENCES[,c('SOTerm', 'DisplayTerm', 'IsProteinCoding')], by = 'SOTerm') %>%
    subset(SOTerm != "") %>%
    dplyr::arrange(., dplyr::desc(.$IsProteinCoding))

labelOrder = context.cnt.combined$DisplayTerm
# context.cnt.combined[['All traits']] = context.cnt.combined[['All traits']] + 0.1
# context.cnt.combined[['CVD']] = context.cnt.combined[['CVD']] + 0.1
labelColors = c('blue', 'green')
tmp <- context.cnt.combined %>%
    reshape2::melt(id.vars = c('SOTerm', 'DisplayTerm', 'IsProteinCoding'))
tmp$color <- factor(tmp$IsProteinCoding) %>% `levels<-`(labelColors) %>% as.character()
p <- context.cnt.combined %>%
    reshape2::melt(id.vars = c('SOTerm', 'DisplayTerm', 'IsProteinCoding')) %>%
    ggplot(mapping =  aes(x = factor(DisplayTerm, levels = labelOrder))) +
        geom_bar(aes(y =value, fill=variable), stat='identity',alpha = 0.8) +
        scale_fill_discrete( guide_legend(title = 'Mapped traits')) +
        scale_y_log10(limits = c(1, NA)) + 
        ylab('Variant count') +
        xlab('Context') +
        theme(legend.position = c(0.83,0.87)) +
        coord_flip()
p
ggsave('../figures/variant_contexts-sorted.png',height = 4, width=6)
p + theme(axis.text.y = element_text(color = tmp$color))
ggsave('../figures/variant_contexts-sorted-colored.png',height = 4, width=6)
```

The ditribution of CVD-associated variants in the human genome reflect the distribution of variants in all variants studied so far 

```{r}
context.cnt.combined %>%
    set_names(c('CONTEXT', 'All_traits', 'CVD')) %>%
    ggplot(aes_string(x = 'All_traits', y = 'CVD')) +
    geom_point() +
    scale_x_log10(limits = c(1, NA)) +
    scale_y_log10(limits = c(1, NA))
# count(gwas, CONTEXT)
# gwas[gwas$CONTEXT == "3_prime_UTR_variant; intron_variant; intergenic_variant",]
# setdiff(gwas.studies$PUBMEDID %>% unique(),gwas$PUBMEDID %>% unique())
# gwas$`DISEASE/TRAIT` %>% unique() %>% length()
# gwas$CONTEXT %>%  unique()  %>% length()
```
#### Distribution of variants in genome, associations filtered at $5\times10^{-8}$

```{r}
gwas.sigVars <-  gwas[gwas$`P-VALUE` < 5e-8,]
context.cnt.all.sigVars =
    gwas.sigVars[is.type1(gwas.sigVars, 'SNPS'), c('SNPS', 'CONTEXT')] %>%
    dplyr::distinct() %>%
    dplyr::count(CONTEXT)

gwas.CVDs.sigVars <- gwas.CVDs[gwas.CVDs$`P-VALUE` < 5e-8,]
context.cnt.CVDs.sigVars =
    gwas.CVDs.sigVars[is.type1(gwas.CVDs.sigVars, 'SNPS'), c('SNPS', 'CONTEXT')] %>%
    dplyr::distinct() %>%
    dplyr::count(CONTEXT)


context.cnt.sigVars = 
    dplyr::full_join(x = context.cnt.all.sigVars,
                     y = context.cnt.CVDs.sigVars,
                     all = TRUE, by = 'CONTEXT', incomparables = 0) %>%
    `[<-`(is.na(.[[2]]), 2, 0) %>%
    `[<-`(is.na(.[[3]]), 3, 0) %>%
    set_names(c('CONTEXT', 'All traits', 'CVD'))
# context.cnt.combined[['All traits']] = context.cnt.combined[['All traits']] + 0.1
# context.cnt.combined[['CVD']] = context.cnt.combined[['CVD']] + 0.1
context.cnt.sigVars %>%
    reshape2::melt(id.vars = 'CONTEXT') %>%
    ggplot(mapping =  aes(x = CONTEXT)) +
        geom_bar(aes(y =value, fill=variable), stat='identity',alpha = 0.8) +
        scale_fill_discrete( guide_legend(title = 'Mapped traits')) +
        scale_y_log10(limits = c(1, NA)) + 
        ylab('Variant count') +
        theme(legend.position = c(0.83,0.87)) +
        coord_flip()

ggsave('../figures/variant_contexts_sigVars.png',height = 4, width=6)
```


### Frequency vs Effect size

```{r}
gwas$EffectSize = gwas$`OR or BETA` %>% as.numeric()
gwas$AlleleFrequency = gwas$`RISK ALLELE FREQUENCY` %>% as.numeric()
gwas[is.type1(gwas, "SNPS"),] %>%
    ggplot(aes(x = AlleleFrequency, y = EffectSize)) +
    scale_y_log10() +
    geom_point()
```

### p-value vs context

```{r}
gwas.CVDs[is.type1(gwas.CVDs, 'SNPS'),c('SNPS', 'CONTEXT', 'PVALUE_MLOG')] %>%
    ggplot(aes(y = PVALUE_MLOG, x = CONTEXT)) +
    geom_boxplot()  +
    coord_flip()
```


### p-value vs calculated impact

Variation effects are predicted using Ensembl VEP, resulting in SIFT and polyphen score

```{r}
# library(xml2)
server <- "https://rest.ensembl.org"
ext <- "/vep/human/id"
var.queries = gwas.CVDs$`STRONGEST SNP-RISK ALLELE`[1:3] %>% jsonlite::toJSON()
var.queries = gwas.CVDs$`SNPS`[1:3] %>% jsonlite::toJSON()
r <- httr::POST(paste(server, ext, sep = ""),
                httr::content_type("application/json"),
                httr::accept("application/json"),
                body = sprintf('{ "ids" : %s }', var.queries))
 
httr::stop_for_status(r)
variants.ensembl = (httr::content(r)) %>% jsonlite::toJSON() %>% jsonlite::fromJSON()
```


```{r}
gwas[grepl(pattern = '\\brs10454444\\b', gwas$SNPS), c('SNPS', 'STRONGEST SNP-RISK ALLELE', 'RISK ALLELE FREQUENCY', 'PUBMEDID', 'STUDY', 'DISEASE/TRAIT', 'INITIAL SAMPLE SIZE', 'REPLICATION SAMPLE SIZE', 'REPORTED GENE(S)', 'MAPPED_GENE', 'PVALUE_MLOG')]
```

```{r}
gwas[gwas$PUBMEDID == '25948407',c('SNPS', 'STRONGEST SNP-RISK ALLELE', 'RISK ALLELE FREQUENCY', 'DISEASE/TRAIT', 'INITIAL SAMPLE SIZE', 'REPLICATION SAMPLE SIZE', 'REPORTED GENE(S)', 'MAPPED_GENE', 'PVALUE_MLOG') ]
```


### Number of genes associated with each disease

```{r}
gwas.queries <- data.table::fread('../data/conditionList.tsv', sep='\t')$query %>% unique() %>%
    lapply(FUN = function(y) {
        gwas[grepl(pattern = y, x = gwas$MAPPED_TRAIT, ignore.case = TRUE) | grepl(pattern = y, x = gwas$`DISEASE/TRAIT`, ignore.case = TRUE),] %>%
            return()
    })

gwas.CVDs = do.call(rbind,gwas.queries)
```

## Myocardial infarction

```{r}
gwas.mi = gwas[grep('[mM]yocardial infarction', gwas$`DISEASE/TRAIT`),] %>%
    `$<-`('DATE', as.Date(.$DATE)) %>%
    `$<-`('RISK_ALLELE_FREQUENCY', as.numeric(.$`RISK ALLELE FREQUENCY`))
ggplot(gwas.mi) +
    geom_point(mapping = aes_string(x='PVALUE_MLOG',y='RISK_ALLELE_FREQUENCY', color='DATE')) +
    theme(legend.position = 'none')
count(gwas.mi, CONTEXT)
```


## Dilated myocardiopathy

### GWAS

```{r}
gwas.dcm = gwas[grep('[dD]ilated [cC]ardiomyopathy', gwas$`DISEASE/TRAIT`),] %>%
    `$<-`('DATE', as.Date(.$DATE))
gwas.dcm$`REPORTED GENE(S)`  %>% sapply(strsplit, split = ',') %>% do.call(c, .) %>% trimws() %>% unique()
```

### Ensembl

```{r}
dcm.dominant = data.table::fread('../data/dilated-cardiomyopathy-dominant-loci-ensembl.csv', sep=',')
dcm.recessive = data.table::fread('../data/dilated-cardiomyopathy-recessive-loci-ensembl.csv', sep=',')
dcm.loci = rbind(dcm.dominant, dcm.recessive) %>%
    `$<-`('inheritance', c(rep('dominant', nrow(dcm.dominant)), rep('recessive', nrow(dcm.recessive))))
    
```

The number of genes associated with
* _Dilated cardiomyopathy, dominant_ is `r dcm.dominant[['Reported gene(s)']] %>% trimws() %>% sapply(strsplit, split = '\r') %>% do.call(c,.) %>% as.character() %>% unique() %>% length()`
* _Dilated cardiomyopathy, recessive is `r dcm.recessive[['Reported gene(s)']] %>% trimws() %>% sapply(strsplit, split = '\r') %>% do.call(c,.) %>% as.character() %>% unique() %>% length()`
* _Dilated cardiomyopathy (both) is `r dcm.loci[['Reported gene(s)']] %>% trimws() %>% sapply(strsplit, split = '\r') %>% do.call(c,.) %>% as.character() %>% unique() %>% length()`

## disgenet

### Mapping disease names to database unique index

```{r}
dgn.all = data.table::fread('../data/disgenet/all_gene_disease_associations.tsv')
dgn.mappings = data.table::fread('../data/disgenet/disease_mappings.tsv', sep='\t')
standardNames = data.table::fread('../data/conditionList.tsv', sep='\t')
```

### All the associations with CVDs found in Disgenet

```{r}
hpo2dgn = standardNames$HPOTermId %>%
    lapply(function(x) {
        return(dgn.mappings[dgn.mappings$code == x, 'diseaseId'])
    }) %>%
    set_names(standardNames$HPOTermId) %>%
    lapply(function(x) {
        subset(dgn.all, diseaseId %in% x[['diseaseId']]) %>%
            `$`('geneId') %>%
            unique() %>%
            length() %>%
            return()
    })

groupColors = c('#a6cee3', '#1f78b4', '#4daf4a', 'darkgreen', '#fb9a99', '#e31a1c', '#fdbf6f', '#ff7f00', '#cab2d6')
tmp = data.frame('HPOTermId' = names(hpo2dgn), 'disgenetCount' = as.numeric(hpo2dgn)) %>%
    merge(standardNames, by = 'HPOTermId', sort = FALSE)
write.table(tmp, file = '../data/geneAssociationCount-disgenet.tsv', sep='\t', quote=FALSE, row.names = FALSE)
tmp$color <- factor(tmp$query) %>% `levels<-`(groupColors) %>% as.character()
tmp %>%
    ggplot() +
    geom_bar(aes(x=factor(HPOTermName, levels=HPOTermName),y=disgenetCount, fill=query),stat = 'identity') +
    theme_bw() +
    theme(axis.text.y = element_text(color = tmp$color)) +
    scale_fill_manual(values = groupColors) +
    guides(fill=guide_legend(title='Group')) +
    ylab('Number of genes on disgenet') +
    xlab('Phenotype') +
    coord_flip()

ggsave('../figures/disgenet-gene-count.png', device = 'png', width = 10, height=4.5)
# standardNames = data.table::fread('../data/conditionList.tsv', sep='\t') %>%
    # merge(dgn.mappings[,c('diseaseId', 'name', 'code', 'vocabularyName')], by.x = 'HPOTermId', by.y= 'code', all.x = TRUE, all.y = FALSE, sort = FALSE)
```



### Gene association similarity among the diseases

```{r}
m = matrix(0, nrow=nrow(dgn.cvdIds),  ncol=nrow(dgn.cvdIds)) %>%
    set_rownames(dgn.cvdIds$name) %>%
    set_colnames(dgn.cvdIds$name)
for (i in 1:nrow(m)) {
    for (j in i:ncol(m)) {
        associations_i = subset(dgn.CVDs, diseaseId %in% dgn.cvdIds[i,'diseaseId']) %>% `$`('geneId') %>% unique()
        if (i == j) m[i,j] = length(associations_i)
        else {
            associations_j = subset(dgn.CVDs, diseaseId %in% dgn.cvdIds[j, 'diseaseId']) %>% `$`('geneId') %>% unique()
            m[i,j] = m[j,i] = intersect(associations_i, associations_j) %>% length()
        }
    }
}

cl = hclust(as.dist(-m))
cl$order
p = plot.heatmap(m[cl$order, cl$order]) +
    theme(axis.text.y = element_text(size = 18),
          axis.text.x = element_text(size = 16),
          legend.title = element_text(size = 18),
          legend.text = element_text(size = 18)) +
    guides(fill = guide_legend('Number of genes')) +
    scale_fill_gradient(low='lightblue', high='darkblue')
p
ggsave(filename = '../figures/disgenet-intersection.png', device = 'png', width = 18, height = 18)
```
